src/test/scala/mesosphere/marathon/raml/PodStatusConversionTest.scala

```diff
@@ -0,0 +1,240 @@
+          ContainerEndpointStatus(name = "web")
+        ),
+        lastUpdated = fixture.since.toOffsetDateTime,
+        lastChanged = fixture.since.toOffsetDateTime
+      )
+    ))
+    status.networks should be('empty)
+  }
+
+  test("ephemeral pod launched, received STARTING status from Mesos") {
+    implicit val clock = ConstantClock()
+    val pod = basicOneContainerPod.copy(version = clock.now())
+
+    clock += 1.seconds
+    val fixture = startingInstance(pod)
+
+    val status = PodStatusConversion.podInstanceStatusRamlWriter((pod, fixture.instance))
+    status.id should be(fixture.instance.instanceId.idString)
+    status.agentHostname should be(Some("agent1"))
+    status.status should be(PodInstanceState.Staging)
+    status.containers should be(Seq(
+      ContainerStatus(
+        name = "ct1",
+        status = "TASK_STARTING",
+        statusSince = fixture.since.toOffsetDateTime,
+        containerId = Some(fixture.taskIds.head.idString),
+        endpoints = Seq(
+          ContainerEndpointStatus(name = "admin", allocatedHostPort = Some(1001)),
+          ContainerEndpointStatus(name = "web")
+        ),
+        lastUpdated = fixture.since.toOffsetDateTime,
+        lastChanged = fixture.since.toOffsetDateTime
+      )
+    ))
+    status.networks.toSet should be(Set(
+      NetworkStatus(Some("dcos"), Seq("1.2.3.4")),
+      NetworkStatus(Some("bigdog"), Seq("2.3.4.5"))
+    ))
+  }
+}
+
+object PodStatusConversionTest {
+  val basicOneContainerPod = PodDefinition(
+    id = PathId("/foo"),
+    containers = Seq(
+      MesosContainer(
+        name = "ct1",
+        resources = Resources(cpus = 0.01, mem = 100),
+        image = Some(Image(kind = ImageType.Docker, id = "busybox")),
+        endpoints = Seq(
+          Endpoint(name = "web", containerPort = Some(80)),
+          Endpoint(name = "admin", containerPort = Some(90), hostPort = Some(0))
+        ),
+        healthCheck = Some(HealthCheck(http = Some(HttpHealthCheck(endpoint = "web", path = Some("/ping")))))
+      )
+    ),
+    networks = Seq(ContainerNetwork(name = "dcos"), ContainerNetwork("bigdog"))
+  )
+
+  case class InstanceFixture(
+    since: Timestamp,
+    agentInfo: Instance.AgentInfo,
+    taskIds: Seq[Task.Id],
+    instance: Instance)
+
+  def createdInstance(pod: PodDefinition)(implicit clock: ConstantClock): InstanceFixture =
+    fakeInstance(pod, InstanceStatus.Created, InstanceStatus.Created)
+
+  def stagingInstance(pod: PodDefinition)(implicit clock: ConstantClock): InstanceFixture =
+    fakeInstance(pod, InstanceStatus.Staging, InstanceStatus.Staging, Some(Protos.TaskState.TASK_STAGING))
+
+  def startingInstance(pod: PodDefinition)(implicit clock: ConstantClock): InstanceFixture =
+    fakeInstance(pod, InstanceStatus.Starting, InstanceStatus.Starting, Some(Protos.TaskState.TASK_STARTING),
+      Some(Map("dcos" -> "1.2.3.4", "bigdog" -> "2.3.4.5")))
+
+  def fakeInstance(
```

Comment:
Ideally we move such functionality in a more general place, since we will use instances in tests all over the place.